### DISCUSSION TITLE
Question: Is rolldown able to remove inconsistencies between development and build completely?

### DISCUSSION THREAD

**Discussion by shuta13** [rolldown/rolldown#452] [Q&A] [answered]
This project seemed good to me! I had a question when I read [the roadmap](https://github.com/rolldown-rs/rolldown/discussions/153). 

I have heard that you are developing rolldown as a possible replacement for vite and resolving inconsistencies in dev and prod. However, even if the tools in the development and production environments are unified by rolldown, if the ESM used in the development environment doesn't work in the production environment as is, it cannot be said that the inconsistencies have been removed. What do you think?

**Comment by hyfdev** [comment]
> even if the tools in the development and production environments are unified by rolldown

Yeah that's just the first step. After the replacing, we could integrate rolldown with vite more deeply, so that we could find solutions to solve the esm problems.

**Reply by shuta13** [reply]
@hyf0 

As you say, it would be the first step in solving the problem! 

Then, if the rolldown team has a specific solution to the problem of ESM inconsistencies, I would like to know. For example, to push for a higher Stage for TS39's [Module Declarations](https://github.com/tc39/proposal-module-declarations) proposal, so that we can use dev's ESM as in prod. To say whether this can be done or not may be difficult since TS39 is involved, but I would like to know what such a policy or you intend to do on its own.

**Reply by hyfdev** [reply, marked as answer]
We are reviewing the [Module Declarations](https://github.com/tc39/proposal-module-declarations) proposal or other non-esm solutions. But it's too early to say what way we gonna use before we reach to [Milestone 4](https://github.com/rolldown-rs/rolldown/discussions/153).

**Reply by shuta13** [reply]
I see. I'm glad to hear the plan for the rolldown development. Thanks.

**Comment by ghiscoding** [comment]
> I have heard that you are developing rolldown as a possible replacement for vite

I believe this sentence is incorrect, it's not a replacement for Vite but rather a replacement for Rollup (hence the name Rolldown). Also, the other thing to note is that again it won't replace Vite but rather be used by Vite in the future. So the dependencies combo of "esbuild+rollup" will be replaced by "rolldown" to be used as a single tool. The end result will be to remove inconsistencies between Dev (esbuild) and Prod (rollup) since Rolldown will be used for both environments. At least that's the long term goal of Rolldown
